enginetest/queries/queries.go

```diff
@@ -7296,7 +7302,8 @@ Select * from (
 		},
 	},
 	{
-		Query: `SELECT JSON_OBJECT(JSON_OBJECT("foo", "bar"), 10);`,
+		SkipServerEngine: true, // the over the wire result need the double quotes to be escaped
```

Comment:
Hmm, the issue for this is on converting the result received from go-sql-driver to dolt sql value. Because we cannot unmarshal the received string `{"{"foo": "bar"}": 10}` to match it with the expected result, it returns nil right now for failing unmarshalling this json string.